lib/neo4j/railtie.rb

```diff
@@ -74,8 +75,10 @@ def final_config!(neo4j_config)
     end
 
     def support_deprecated_session_configs!(neo4j_config)
-      ActiveSupport::Deprecation.warn('neo4j.config.sessions is deprecated, please use neo4j.config.session (not an array)') if neo4j_config.sessions.present?
-      neo4j_config.session ||= (neo4j_config.sessions && neo4j_config.sessions[0]) || {}
+      if neo4j_config.sessions.present?
```

Comment:
No, I meant `sessions`. The 2nd replaced line makes sense only under the above condition. Previously this line contained a lazy initialization as a side effect, but in a wrong place. We cannot depend on this deprecation method to initialize the session. That's the reason for default parameter in the `setup!` method.